### PR TITLE
chore: remove date-fns

### PR DESCRIPTION
**Pull request by dkarnutsch**
This PR removes date-fns completely. Orphaned package react-dates is also moved to react-admin-date-picker. Unmet peer-dependency react-with-direction is added.

Breaking Changes:
- `useLocaleName();` is removed and should be replaced with `const intl = useIntl(); const localeName = intl.locale;`
- `LocaleContextProvider` is removed and should be replaced with `IntlProvider`
- `FormatLocalized` is removed and should be replaced with https://formatjs.io/docs/react-intl/components#formatteddate and https://formatjs.io/docs/react-intl/components#formattedtime

### CODE REVIEW

File: packages/react-admin-date-picker/src/DateRangePicker.tsx [hunk 1]
```diff
@@ -31,7 +31,9 @@ export const FinalFormDateRangePicker: React.FC<IProps> = ({
     endPlaceholder,
     ...props
 }) => {
-    const localeName = useLocaleName();
+    const intl = useIntl();
+    const localeName = intl.locale;
```
Comment (rabengraph) [inline packages/react-admin-date-picker/src/DateRangePicker.tsx:35]: useLocaleName() is interchangeable with intl.locale!
I am impressed, this is some neat refactoring.

**Review by rabengraph — APPROVED** [review]

**Comment by rabengraph** [comment]
The "breaking-changes" statements from the PR-description should be committed to code within this PR (into packages/react/admin/changelog.md)
I think it's a good idea to create the changelog.md now.

File: packages/react-admin-date-picker/package.json [hunk 2]
```diff
@@ -7,9 +7,9 @@
         "lib"
     ],
     "dependencies": {
-        "date-fns": "^2.0.0-alpha.37",
         "moment": "^2.24.0",
-        "react-dates": "^21.3.2"
+        "react-dates": "^21.3.2",
+        "react-with-direction": "^1.3.1"
```
Comment (nsams) [inline packages/react-admin-date-picker/package.json:12]: what is this?

Comment (dkarnutsch) [reply packages/react-admin-date-picker/package.json:12]: An unmet peer dependency of react-dates (see: https://github.com/airbnb/react-dates/blob/master/package.json).

**Review by nsams — APPROVED** [review]

**Pull request merged**